libcst/codemod/visitors/_apply_type_annotations.py

```diff
@@ -593,6 +637,60 @@ def _insert_empty_line(
             *statements[1:],
         ]
 
+    def _match_signatures(  # noqa: C901: Too complex
```

Comment:
I wonder if we might want to separate out structural compatiblity from type annotation compatibility.

I'll explain more in the tests